File: Tutoriels/CSharp/Projets/src/manager/GProcessPrint.cs
//===============================================
using System;
//===============================================
public class GProcessPrint : GProcess {
    //===============================================
    private static GProcessPrint m_instance = null;
    //===============================================
    private GProcessPrint() {

    }
    //===============================================
    new public static GProcessPrint Instance() {
        if(m_instance == null) {
            m_instance = new GProcessPrint();
        }
        return m_instance;
    }
    //===============================================
    public override void Run(string[] args) {
        GPrint.Instance().Print("Bonjour tout le monde");
    }
    //===============================================
}
//===============================================

File: Tutoriels/Python/Projets/src/manager/GProcessPrint.py
#================================================
from GProcess import GProcess
from GPrint import GPrint
#================================================
class GProcessPrint(GProcess):
    #================================================
    m_instance = None
    #================================================
    def __init__(self):
        if GProcessPrint.m_instance == None:
            GProcessPrint.m_instance = self        
    #================================================
    @staticmethod
    def Instance():
        if GProcessPrint.m_instance == None:
            GProcessPrint()
        return GProcessPrint.m_instance
    #================================================
    def run(self):
        GPrint.Instance().printData("Bonjour tout le monde")
        GPrint.Instance().printData("Gerard KESSE", "NOM")
#================================================
